src/emu/dinvram.h

```diff
@@ -38,15 +46,43 @@ class device_nvram_interface : public device_interface
 	void nvram_load(emu_file &file) { nvram_read(file); }
 	void nvram_save(emu_file &file) { nvram_write(file); }
 
+	bool has_battery_backup() const { return device_has_battery_backup(); }
+
+	memory_array &memarray() { return m_memarray; }
+
 protected:
 	// derived class overrides
 	virtual void nvram_default() = 0;
-	virtual void nvram_read(emu_file &file) = 0;
-	virtual void nvram_write(emu_file &file) = 0;
+	virtual void nvram_read(emu_file &file);
+	virtual void nvram_write(emu_file &file);
+	virtual bool device_has_battery_backup() const { return true; }
+
+private:
+	memory_array m_memarray;
```

Comment:
This very much doesn't belong in this class.  If you think it's really worthwhile, make a class that derives from `device_nvram_interface` to hold the `memory_array` and change devices that use it to derive from that.